src/caffeine/caffeine.c

```diff
@@ -157,9 +137,61 @@ void c_co_broadcast(CFI_cdesc_t * a_desc, int source_image, int* stat, int num_e
      size_t c_sizeof_a = a_desc->elem_len;
      int nbytes = num_elements * c_sizeof_a;
   
-     gex_Event_t ev
-       = gex_Coll_BroadcastNB(myteam, source_image-1, c_loc_a, c_loc_a, nbytes, 0);
-     gex_Event_Wait(ev);  
+  int data_type = a_desc->type;
+
+  gex_Event_t ev
+    = gex_Coll_BroadcastNB(myteam, source_image-1, c_loc_a, c_loc_a, nbytes, 0);
+  gex_Event_Wait(ev);  
+
+  if (stat != NULL) *stat = 0;
+}
+
+void c_co_sum(CFI_cdesc_t* a_desc, int result_image, int* stat, char* errmsg, size_t num_elements)
+{
+  CFI_type_t a_type = a_desc->type;
+  size_t c_sizeof_a;
+
+  switch (a_type)
+  {
+    case CFI_type_int32_t:        a_type = GEX_DT_I32; break;
```

Comment:
`gex_DT_t` is not guaranteed to be any particular type and in general will not match `CFI_type_t`, which means the assignment above only happens to compile.

Please declare a second variable of type `gex_DT_t` and use that to hold GEX_DT constants.

Reply:
Note to self: use two different variables: `a_type` and something to hold the `GEX_DT_*` value.  The latter should be of type `gex_DT_t`.

Reply:
fixed by fe23297dc64045e934160a328a03a38afb3eba59